_posts/news/2020-10-16-announcing-scalajs-1.3.0.md

```diff
@@ -0,0 +1,322 @@
+* Module initializers: Code that gets executed when a module is imported.
+
+The Scala.js linker determines how to group entry points into different (public) modules by using their assigned `moduleID`.
+The default `moduleID` is `"main"`.
+
+The `moduleID` of a top level export can be specified using the `moduleID` parameter.
+The `moduleID` of a `ModuleInitializer` can be specified by the `withModuleID` method.
+
+**Example**:
+
+Say you have the following `App.scala` and `build.sbt`:
+
+{% highlight scala %}
+package my.app
+
+import scala.collection.mutable
+import scala.scalajs.js.annotation._
+
+// Separate objects to allow for splitting.
+
+object AppA {
+  @JSExportTopLevel(name = "start", moduleID = "a")
+  def a(): Unit = println("hello from a")
+}
+
+object AppB {
+  private val x = mutable.Set.empty[String]
+
+  @JSExportTopLevel(name = "start", moduleID = "b")
+  def b(): Unit = {
+    println("hello from b")
+    println(x)
+  }
+
+  def main(): Unit = x.add("something")
+}
+{% endhighlight %}
+
+{% highlight scala %}
+import org.scalajs.linker.interface.ModuleInitializer
+
+scalaJSModuleInitializers in Compile += {
+  ModuleInitializer.mainMethod("my.app.AppB", "main").withModuleID("b")
+}
+{% endhighlight %}
+
+This would generate two public modules `a.js` / `b.js`.
+`a.js` will export a method named `start` that calls `AppA.a`.
+`b.js` will export a method named `start` that calls `AppB.b`.
+Further, importing `b.js` will call `AppB.main`.
+
+Note that there is no public module `main.js`, because there is no entry point using the default `moduleID`.
+
+### Module Split Styles
+
+So far, we have seen how public modules can be configured.
+Based on the public modules, the Scala.js linker generates internal modules for the shared code between the public modules.
+Unlike public modules, internal modules may not be imported by user code.
+Doing so is undefined behavior and subject to change at any time.
+
+The linker genereates internal modules automatically based on the dependency graph of the code and `moduleSplitStyle`.
+You can change it as follows:
+
+{% highlight scala %}
+import org.scalajs.linker.interface.ModuleSplitStyle
+scalaJSLinkerConfig ~= (_.withModuleSplitStyle(ModuleSplitStyle.SmallestModules))
+{% endhighlight %}
+
+There are currently two module split styles: `FewestModules` and `SmallestModules`.
+
+#### `FewestModules`
+
+Create as few modules as possible without including unnecessary code.
+This is the default.
+
+In the example above, this would generate:
+
+* `a.js`: public module, containing `AppA` and the export of `start`.
+* `b.js`: public module, containing `AppB`, `mutable.Set`, the export of `start` and the call to `AppB.main`
+* `a-b.js`: internal module, Scala.js core and the implementation of `println`.
+
+This also works for more than two public modules, creating intermediate shared (internal) modules as necessary.
+
+#### `SmallestModules`
```

Comment:
It might be good to add a sentence saying why this can be useful? (i.e., to speed up incremental bundling if the output is fed to downstream tooling)